types/react-select/index.d.ts

```diff
@@ -7,7 +7,7 @@
 //                 Daniel Del Core <https://github.com/danieldelcore>
 //                 Joonas Rouhiainen <https://github.com/rjoonas>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.9
+// Minimum TypeScript Version: 3.4
```

Comment:
Why do we need a minimum TypeScript version? Can this be removed?

Reply:
Yeah I think something to do with the generics. CI said to update the header with this

Reply:
Can you try reverting it again? It might pass with some of the changes you've made since then. If not, I can look into more of why it's failing.

Reply:
reverted

Reply:
Odd, it still fails. Maybe they changed something about the build, since it's complaining about a file that you didn't change? Looking into it.

Reply:
I think the error is legitimate. It's complaining about what we're talking about it the other comment thread, so I think it will be resolved once we resolve the other issue.

Reply:
Here's the error pulled from CI logs, hopefully be al little easier to dissect.

```
1> Error: /home/runner/work/DefinitelyTyped/DefinitelyTyped/types/react-select/test/examples/CreateFilter.tsx:43:11
64
1> ERROR: 43:11  expect  Compile error in typescript@3.3 but not in typescript@3.4.
65
1> Fix with a comment '// Minimum TypeScript Version: 3.4' just under the header.
66

67
1> Type 'ColourOption' is not assignable to type 'ValueType<Option>'.
68

69
1>   Property 'data' is missing in type 'ColourOption' but required in type 'Option'.
70
1> ERROR: 47:11  expect  Compile error in typescript@3.3 but not in typescript@3.4.
71
1> Fix with a comment '// Minimum TypeScript Version: 3.4' just under the header.
72
1> Type 'ReadonlyArray<ColourOption>' is not assignable to type 'ReadonlyArray<Option> | ReadonlyArray<GroupType<Option>> | undefined'.
73
1>   Type 'ReadonlyArray<ColourOption>' is not assignable to type 'ReadonlyArray<Option>'.
74
1>     Type 'ColourOption' is not assignable to type 'Option'.
75
1> 
76
1>     at /home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/dtslint/bin/index.js:206:19
77
1>     at Generator.next (<anonymous>)
78
1>     at fulfilled (/home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/dtslint/bin/index.js:6:58)
```